src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -800,7 +800,8 @@ public Task<HtmlString> RenderSectionAsync(string name, bool required)
             else if (required)
             {
                 // If the section is not found, and it is not optional, throw an error.
-                throw new InvalidOperationException(Resources.FormatSectionNotDefined(sectionName, Path));
+                var message = Resources.FormatSectionNotDefined(sectionName, ViewContext.View.Path);
```

Comment:
Oops, this one is strange. Looked at the bug and see we _want_ to use `View.Path` here. That's unusual but I can see the reasoning.